src/communication/xcConnection.ts

```diff
@@ -31,12 +31,14 @@ Connection.prototype.init = function (xcApiFileName, serverUrl, sessionData, ses
     let session = SessionFactory.create(serverUrl, null, sessionData);
     let getXcApiRequest = function (xcApiFileName, sessionListener) {
         session.privateSubscriber.getXcApi(xcApiFileName, function (xcApi) {
-            let parser = new Parser();
-            let configuration = new Configuration(parser);
-            configuration.init(xcApi);
-            session.configuration = configuration;
-            session.replyPublisher.configuration = configuration;
-            sessionListener(null, session);
+            const parser = new DefaultApiConfigurationParser();
+            const configurationPromise = parser.parse(xcApi);
+            configurationPromise.then(configuration => {
```

Comment:
that's why I think we don't need a promise to parse the configuration, since they come from the server anyway.

I would expect getXcApi to return a promise, and we would

``getXcpi().then(configFile=> session.config = parser.parse(configFile))``